src/get_equality.jl

```diff
@@ -101,12 +76,38 @@ types of constraints in [`is_equality`](@ref) and
 `set_implies_equality` should be defined.
 
 """
-function set_implies_equality(
-    set::T
-)::Bool where T<:MOI.AbstractSet
+function set_implies_equality(set::MOI.AbstractSet)::Bool
+    return false
+end
+
+function set_implies_inequality(set::MOI.GreaterThan)::Bool
+    return true
+end
+
+function set_implies_inequality(set::MOI.LessThan)::Bool
+    return true
+end
+
+function set_implies_inequality(
+    set::MOI.Interval;
+    tolerance::Float64=0.0,
+)::Bool
+    # NOTE: tolerance has not been implemented in any calling function
+    return abs(set.upper - set.lower) > tolerance
+end
+
+function set_implies_inequality(set::MOI.AbstractSet)::Bool
     return false
 end
 
+function set_implies_inequality(set::MOI.AbstractVectorSet)::Bool
+    throw(TypeError(set, MOI.AbstractScalarSet, typeof(set)))
+end
+
+# TODO: set_implies_inequality(set::MOI.AbstractSet, tolerance)
+# that just calls set_implies_inequality(set). I.e. the default
+# is to ignore the tolerance.
```

Comment:
This is probably what we will need to do to pass `tolerance` to `set_implies_inequality(set::MOI.Interval)`